src/StorageEngines/ForestDB/storage.forestdb.shared/src/ForestDBCouchStore.cs

```diff
@@ -1260,8 +1253,35 @@ public void SetInfo(string key, string info)
             var success = RunInTransaction(() =>
             {
                 try {
+                    
                     var docId = inDocId;
                     var prevRevId = inPrevRevId;
+
+                    // https://github.com/couchbase/couchbase-lite-net/issues/749
+                    // Need to ensure revpos is correct for a revision inserted on top
+                    // of a deletion
+                    var existing = (C4Document*)ForestDBBridge.Check(err => Native.c4doc_getForPut(Forest, docId, prevRevId?.ToString(), deleting,
+                        allowConflict, err));
+
+                    if(existing->IsDeleted) {
+                        var attachments = properties.CblAttachments();
+                        if(attachments != null) {
```

Comment:
Since this test is cheap and is likely to fail (most docs don't have attachments), it would be best to do it before loading the old revision (i.e. before line 1263.) That will minimize the performance hit.